hack/ensure-kind.sh

```diff
@@ -46,7 +46,13 @@ verify_kind_version() {
   fi
 
   local kind_version
-  kind_version="v$(kind version -q)"
+  if [ -x "$(command -v kind)" ]; then
+    kind_version="v$(kind version -q)"
+  else
+    echo "error: kind installed, but GOPATH_BIN=${GOPATH_BIN} not in your PATH"
```

Comment:
Some of the other binary install scripts assume `GOPATH_BIN` is in the PATH (eg. kubectl https://github.com/kubernetes-sigs/cluster-api/blob/ff6896457dc58c84de57e80d60c3125d200ae9e2/hack/ensure-kubectl.sh), I wonder if there is a way to make this solution more generic, maybe by extracting this logic to a `ensure-gopathbin.sh` script to check that GOPATH_BIN is part of PATH at the beginning of every hack/ binary install script?

Reply:
We have `ensure-go.sh`, which is called before `ensure-kubectl.sh` and `ensure-kind.sh` in `ci-e2e.sh`, and is thus expected to be run before others. I could check the `GOPATH_BIN` exists and is in the `PATH` there, and we could drop some logic from other scripts?

Reply:
I would really prefer if we don't depend on the scripts being called in a certain order. This can change and is thus very brittle.

It would be better if every script checks it, either by calling a "shared script" or simple copy&paste would be fine for me as well.

Reply:
This basically leaves copy&paste or a function in `utils.sh` as options. Since its going to be called in three places, I'd lean towards `utils.sh`.